data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -135,7 +136,11 @@ public void start(Buffer<Record<Event>> buffer) {
                             break;
                         case PLAINTEXT:
                         default:
-                            kafkaConsumer = new KafkaConsumer<String, String>(consumerProperties);
+                            if (sourceConfig.getSchemaConfig().getType() == SchemaRegistryType.AWS_GLUE) {
+                                kafkaConsumer = new KafkaConsumer(consumerProperties, stringDeserializer, glueDeserializer);
+                            } else {
+                                kafkaConsumer = new KafkaConsumer<String, String>(consumerProperties);
```

Comment:
nit: we can have a common way of creating KafkaConsumer, which will be to take Deserializer as an arguement.

Reply:
I don't want to such refactoring now causing any unforeseen issues. Will do after the release.